src/utils/element.ts

```diff
@@ -47,55 +47,66 @@ export function getPathValuePairs(object: object): FlatObject {
   return flatFSHObject;
 }
 
+const typeCache: Map<string, Map<string, string>> = new Map();
+
 export function getFSHValue(
-  key: string,
-  flatObject: FlatObject,
+  index: number,
+  flatArray: [string, string | number | boolean][],
   resourceType: string,
   fisher: utils.Fishable
 ): number | boolean | string | fshtypes.FshCode | bigint {
-  const value = flatObject[key];
+  const [key, value] = flatArray[index];
   const fishingType = resourceType === 'Concept' ? 'CodeSystem' : resourceType;
-  const definition = fhirtypes.StructureDefinition.fromJSON(
-    fisher.fishForFHIR(fishingType, utils.Type.Resource, utils.Type.Type)
-  );
   // Finding element by path works without array information and _ from children of primitives
   let pathWithoutIndex = removeUnderscoreForPrimitiveChildPath(key).replace(/\[\d+\]/g, '');
   if (resourceType === 'Concept') {
     pathWithoutIndex = `concept.${pathWithoutIndex}`;
   }
-  const element = definition.findElementByPath(pathWithoutIndex, fisher);
+  // If we have already looked up this path for this resource, get it from cache
+  const type = typeCache.get(resourceType)?.get(pathWithoutIndex);
+  if (type === 'code') {
+    return new fshtypes.FshCode(value.toString());
+  } else if (type === 'integer64') {
+    return BigInt(value);
+  } else if (type) {
+    return value;
+  }
+
   // If the path is one on an entry/contained resource, find the element on the ResourceType of the entry/contained resource
-  if (
-    element == null &&
-    (pathWithoutIndex.startsWith('entry.resource.') || pathWithoutIndex.startsWith('contained.'))
-  ) {
-    let basePath: string;
-    if (element == null && pathWithoutIndex.startsWith('entry.resource.')) {
-      basePath = key.split('.').slice(0, 2).join('.') + '.';
-    } else if (element == null && pathWithoutIndex.startsWith('contained.')) {
-      basePath = key.split('.').slice(0, 1).join('.') + '.';
+  if (pathWithoutIndex.startsWith('entry.resource.') || pathWithoutIndex.startsWith('contained.')) {
+    const [, baseKey, newKey] = key.match(/^(entry\[\d+\].resource|contained\[\d+\])\.(.+)/);
+    const nestedResourceStartIndex = flatArray.findIndex(([key]) => key.startsWith(baseKey));
+    let nestedResourceEndIndex = nestedResourceStartIndex + 1;
```

Comment:
It might be helpful to put a small comment in here explaining that this strategy is safe because the nested resource indices are guaranteed to be sequential.